chaoslib/openebs/cstor_replica_network_delay.yaml

```diff
@@ -0,0 +1,163 @@
+  register: connected_replica
+  until: "connected_replica.stdout == \"2\""
+  delay: 30
+  retries: 10
+
+- name: Get current iostats from target
+  shell: kubectl exec -it {{ istgt_replica.stdout }} -n {{ operator_ns }} --container cstor-istgt -- istgtcontrol -q iostats |json_pp| grep WriteIOPS |awk -F ':' '{print $2}' | tr -d -c 0-9
+  register: last_iostats
+
+- name: Generate database name
+  shell: cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1
+  register: new_db_name
+
+- name: Create new database in the mysql
+  include_tasks: "/common/utils/mysql_data_persistence.yml"
+  vars:
+    status: 'LOAD'
+    #ns: "app-percona-ns"
+    #pod_name: "percona-786c6b7c7f-7qkxm"
+    ns: "{{ app_ns }}"
+    pod_name: "{{ app_pod.stdout }}"
+    dbuser: 'root'
+    dbpassword: 'k8sDem0'
+    dbname: "{{ new_db_name.stdout }}"
+
+- name: Check if target got any new write IOs
+  shell: kubectl exec -it {{ istgt_replica.stdout }} -n {{ operator_ns }} --container cstor-istgt -- istgtcontrol -q iostats |json_pp| grep WriteIOPS |awk -F ':' '{print $2}' | tr -d -c 0-9
+  register: iostats
+  retries: 10
+  delay: 3
+  until: iostats.stdout != last_iostats.stdout
+
+- name: Remove packet loss rule from targeted replica pod
+  shell: kubectl exec -it {{ targeted_replica_pod }} -n {{ operator_ns }} --container cstor-pool -- bash -c "tc qdisc del dev eth0 root netem loss 100.00"
+  register: apt_rc
+
+- name: Wait until all replica gets connected
+  shell: kubectl exec -it {{ istgt_replica.stdout }} -n {{ operator_ns }} --container  cstor-istgt -- istgtcontrol -q replica |json_pp |grep "\"replicaId" |awk -F ':' '{print $2}' | tr -d ',' |wc -l
+  register: connected_replica
+  until: "connected_replica.stdout == \"3\""
+  delay: 30
+  retries: 10
+
+- name: Wait until all replica becomes healthy
+  shell: kubectl exec -it {{ istgt_replica.stdout }} -n {{ operator_ns }} --container  cstor-istgt -- istgtcontrol -q replica |json_pp |grep "\"status" |awk -F ':' '{print $2}' | tr -d ','
+  register: replica_status
+  until: replica_status.stdout.find("Degraded") == -1
+  retries: 50
+  delay: 10
+
+- name: Generate snapshot name
+  shell: cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1
+  register: snapname
+
+- set_fact:
+    snap_name: "{{ snapname.stdout }}"
+
+- name: Take snapshot for data verification
+  shell: kubectl exec -it {{ istgt_replica.stdout }} -n {{ operator_ns }} --container  cstor-istgt -- istgtcontrol snapcreate {{ pv.stdout }} {{ snap_name }} 30 30
+  register: snap_status
+
+- block:
+    - name: Install netcat on current host
+      shell: apt-get update && apt-get -y install netcat iproute2
+      register: dataset
+
+    - include: fetch_data_from_replica.yml
+      loop: "{{ cstor_replicas_pod }}"
+      loop_control:
+        loop_var: rpod
+
+    - set_fact:
+        base_replica: "{{ cstor_replicas_pod | list | first }}"
```

Comment:
how can we be sure if this is the first replica always ?

Reply:
Here, replica selection is not fixed. We are selecting any first replica and blocking IOs on it.